pandas/core/algorithms.py

```diff
@@ -467,6 +467,8 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
     elif is_extension_array_dtype(comps.dtype) or is_extension_array_dtype(
         values.dtype
     ):
+        if type(comps).__name__ == "IntegerArray":
```

Comment:
You want to check isinstance(comps, numpy.ma.MaskedArray) I think

Reply:
Maybe not the numpy function but check for the masked EA class

Reply:
So we could use
```Python
if isinstance(comps, BaseMaskedArray):
```
here.